documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc

```diff
@@ -0,0 +1,54 @@
+instead the application server will automatically enable Hibernate support if it detects that your application works with JPA/Hibernate.
+
+There may be times though where a newer version of Hibernate ORM is available than the one coming with a given WildFly release.
+For that case the Hibernate ORM project provides a ZIP file containing the required modules to update a WildFly installation to newer versions.
+The module ZIP file is available from SourceForge and BinTray, alike the ZIP/TAR.GZ distributions (TODO: Add link).
+As of Hibernate ORM 5.1.1, a ZIP file for upgrading WildFly 10 to that Hibernate version (from 5.0.x) is provided.
+The general goal is to provide a ZIP file for bringing new versions of Hibernate ORM to the WildFly version current at the time of the Hibernate release.
+
+Once downloaded, extract the contents of the ZIP file into the _modules_ directory of your WildFly installation
+(shut down the application server before, should it be running).
+Note that the Hibernate ORM modules coming with WildFly will remain untouched,
+i.e. you can switch between the original version and the new version from the ZIP file as needed as a matter of configuration.
+
+The module system of the application server uses what's called "slots" to differentiate between several versions of one module.
+By default, slot "main" of the Hibernate ORM modules will be used to provide JPA support for given deployments (representing the Hibernate version coming with WildFly itself).
+In order to use another slot, specify the following two property in the _persistence.xml_ file of your application:
+
+[[wildfly-using-custom-hibernate-orm-version]]
+.Using a specific version of Hibernate ORM
+====
+[source, XML]
+----
+<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence
+    http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd"
+    version="2.1" >
+
+    <persistence-unit name="examplePu" transaction-type="JTA">
+
+        <!-- ... -->
+
+        <properties>
+            <property name="jboss.as.jpa.providerModule" value="org.hibernate:<%VERSION%>"/>
+        </properties>
+
+        <!-- ... -->
+
+    </persistence-unit>
+</persistence>
+----
+====
+
+For `<%VERSION%>` specify the version of the module ZIP you downloaded, e.g. `5.1.1.Final`.
+Alternatively you can specify just the minor version, e.g. `5.1`.
+That way you can update your WildFly server with further micro updates of the same Hibernate ORM release family as they are released,
+without having to adapt your _persistence.xml_ upon each micro update.
+Note that you can have several micro updates of the same release family next to each other within a WildFly instance
+and switch between them by means of the properties shown above.
+If you are specifying just the minor version as module slot, the modules from the ZIP file you unpacked last will be used.
```

Comment:
Looks great. Do you think we need to point out that these modules are intended to work with WildFly 10?

Maybe we should clarify that we'll be releasing such "upgrade packs" for selected AS editions, and that people should make sure to take "the right one": not any random combination will necessarily work, nor be an option.